File: Online-Judges/Codeforces/Python/1101-E.py
n=int(input())
maior=menor=0;
for i in range(n):
    op,x,y=input().split()
    x=int(x); y=int(y)
    
    if op=="+":
        if maior<max(x, y):
            maior=max(x, y)
        if menor<min(x, y):
            menor=min(x, y)
    else:
        if maior<=max(x, y) and menor<=min(x, y):
            print("YES")
        else:
            print("NO")

File: Online-Judges/Codeforces/Cpp/1101-E.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
    string op;
    cin.sync_with_stdio(false);
    cin.tie(NULL);
    int n, menor, maior, x, y;
    cin>>n;
    menor=maior=0;
    for(int i=0; i<n; i++){
        cin>>op>>x>>y;
        if(op=="+"){
            if(maior<max(x, y))
                maior=max(x, y);
            if(menor<min(x, y))
                menor=min(x, y);
        }else{
        if (maior<=max(x, y) && menor<=min(x, y))
            printf("YES\n");
        else
            printf("NO\n");
        }
    }
    return 0;
}

